ui/apps/platform/src/Containers/Collections/CollectionFormModal.tsx

```diff
@@ -60,75 +73,94 @@ function CollectionsFormModal({
 
     let content: ReactElement | null = null;
 
+    let modalTitle = '';
+
     if (error) {
         content = (
-            <>
-                {error.message}
-                {/* TODO - Handle UI for network errors */}
-            </>
+            <Bullseye className="pf-u-p-2xl">
+                <CollectionLoadError error={error} />
+            </Bullseye>
         );
+        modalTitle = 'Collection error';
     } else if (loading) {
-        content = <>{/* TODO - Handle UI for loading state */}</>;
+        content = (
+            <Bullseye className="pf-u-p-2xl">
+                <Spinner isSVG />
+            </Bullseye>
+        );
+        modalTitle = 'Loading collection';
     } else if (data) {
         content = (
-            <Modal
```

Comment:
This change looks large, but it is primarily two simple changes:

- Extract the `<Modal>` wrapper so that the three content states can be passed as children
- Conditionally render the header button controls in the case of a loading or error state